File: pkg/manifest/validate_test.go
package manifest

import (
	"errors"
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
)

var _ = Context("Validate", func() {
	It("should call validator.Validate w/ expected args & return result", func() {
		/* arrange */
		providedPkgPath := "dummyPkgPath"

		expectedErrs := []error{
			errors.New("dummyErr1"),
			errors.New("dummyErr2"),
		}
		fakeValidator := new(fakeValidator)

		fakeValidator.ValidateReturns(expectedErrs)

		objectUnderTest := _Manifest{
			validator: fakeValidator,
		}

		/* act */
		actualErrs := objectUnderTest.Validate(providedPkgPath)

		/* assert */
		Expect(fakeValidator.ValidateArgsForCall(0)).To(Equal(providedPkgPath))
		Expect(actualErrs).To(Equal(expectedErrs))
	})
})

File: sdks/js/src/opspec/interpreter/opcall/params/param/socket/validate.test.ts
import objectUnderTest from './validate'

describe('validate', () => {
  describe('value falsy', () => {
    it('returns expected result', () => {
      /* arrange */
      const expectedResult = [new Error('socket required')]

      /* act */
      const actualResult = objectUnderTest('')

      /* assert */
      expect(actualResult).toEqual(expectedResult)
    })
  })
  describe('value truthy', () => {
    it('returns expected result', () => {
      /* arrange/act */
      const actualResult = objectUnderTest('some socket')

      /* assert */
      expect(actualResult).toEqual([])
    })
  })
})
